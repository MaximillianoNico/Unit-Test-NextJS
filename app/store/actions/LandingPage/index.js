import { ActionType } from '../../index';

export const historyUser = res => { 
    return {
        type: ActionType.HistorySearch, payload: res 
    }
};