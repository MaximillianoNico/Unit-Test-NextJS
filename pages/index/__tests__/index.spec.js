/* eslint-disable no-undef */
import { shallow, mount } from 'enzyme'
// import toJson from 'enzyme-to-json'

import Homepage from '../index';
import { historyUser } from '../../../app/store/actions/LandingPage';

describe("Unit Test", ()=>{
    describe("Component <Index />", () => {
        it("should be definded", () =>{
            expect(Homepage).toBeDefined();
        })
        it("should match to snapshot", () => {
            const wrapper = shallow(<Homepage/>)
            expect(wrapper).toMatchSnapshot();
        })
        it('should be render form', () => {
            const wrapper = mount(<Homepage/>).find(
                <div className="form-group__container">
                    <input className=""/>
                </div>
            )
            expect(wrapper).toBeDefined()
        })
    })

    describe("Redux Actions Landing page ",() => {
        it("should update history reducer", () => {
            const data = [{data: '1'},{data: '2'}]
            const wrapperRedux = historyUser(data);

            const expectPayload = {
                payload: data,
                type: 'HISTORY_SEARCH'
            }
            expect(wrapperRedux).toEqual(expectPayload)
        })
    })
});