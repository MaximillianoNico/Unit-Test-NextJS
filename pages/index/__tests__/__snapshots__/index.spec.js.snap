// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Test Component <Index /> should match to snapshot 1`] = `
<div>
  Landing Page 
  Unit Test
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <div
      className="form-group__container"
    >
      <input
        className=""
      />
    </div>
  </div>
</div>
`;
